File: multiply_without_loops.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

int multiplyWithoutOperator(int a, int b)
{
    if(a == 0 || b == 0)
        return 0;

    if (a > 0)
        return (b + multiplyWithoutOperator(a - 1, b));
    else
        return -multiplyWithoutOperator(-a, b);
}

int main (int argc, char *argv[])
{
    int a;
    int b;
    int result;
    cout << "Enter a first number a = ";
    cin >> a;
    cout << "Enter a second number b = ";
    cin >> b;

    if(cin.fail())
    {
        cout << "--ERROR: a or b is not a integer! Try  again." << endl;
    }
    else
    {
        result = multiplyWithoutOperator(a, b);
        cout << a << " * " << b << " = " << result << endl;
    }

}

File: multiply_without_loops.py
def multiply_without_operator(a, b):
    if a == 0 or b == 0:
        return 0

    if a > 0:
        return b + multiply_without_operator(a - 1, b)
    else:
        return -multiply_without_operator(-a, b)


def main():
    try:
        a = int(input('Enter a first number a = '))
        b = int(input('Enter a second number b = '))
        result = multiply_without_operator(a, b)
        print("{} * {} = ".format(a, b), result)
    except ValueError as e:
        print("--ERROR: a or b is not a integer! Try  again.")
        print(e)

if __name__ == "__main__":
    main()
